api/api_test.go

```diff
@@ -347,18 +412,32 @@ func TestFailedToAddFilterBlueprintDimensionOption(t *testing.T) {
 		So(response, ShouldResemble, "Bad request\n")
 	})
 
-	Convey("When a dimension for filter blueprint does not exist, a not found status is returned", t, func() {
+	Convey("When the filter blueprint is unpublished, and the request is unauthenticated, a not found status is returned", t, func() {
 		r, err := http.NewRequest("POST", "http://localhost:22100/filters/12345678/dimensions/age/options/33", nil)
 		So(err, ShouldBeNil)
 
 		w := httptest.NewRecorder()
-		api := routes(authHeader, host, mux.NewRouter(), &mocks.DataStore{NotFound: true}, &mocks.FilterJob{}, &mocks.DatasetAPI{}, previewMock)
+		api := routes(authHeader, host, mux.NewRouter(), &mocks.DataStore{Unpublished: true}, &mocks.FilterJob{}, &mocks.DatasetAPI{Unpublished: true}, previewMock)
+		api.router.ServeHTTP(w, r)
+		So(w.Code, ShouldEqual, http.StatusNotFound)
+
+		bodyBytes, _ := ioutil.ReadAll(w.Body)
+		response := string(bodyBytes)
+		So(response, ShouldResemble, "Filter blueprint not found\n")
+	})
+
+	Convey("When a dimension for filter blueprint does not exist, a not found status is returned", t, func() {
```

Comment:
The test title does not marry up with assertion, status code is 400 but title assumes it will be 404?